### DISCUSSION TITLE
global onSuccess : get query informations

### DISCUSSION THREAD

**Discussion by olup** [TanStack/query#632] [General]
Global onSuccess let's you have the result datas. Any way to also know details about the query that was successful (like it's key, for persistence cache updates) ?

**Comment by wolverineks** [comment]
you could look into subscribing to individual queries
```ts
const query = queryCache.getQuery(queryKey)
query.subscribe(queryState => {
  if (queryState.status === 'success') {
    // do the thing
  }
})
```
